src/app/features/genes/components/gene-nominations/gene-nominations.component.ts

```diff
@@ -36,28 +36,33 @@ export class GeneNominationsComponent {
     }
 
     this.teamService.getTeams().subscribe((res: TeamsResponse) => {
-      const nominations: NominatedTarget[] = [];
       const teams: Team[] = res.items;
-      const teamNames = this._gene?.nominatedtarget?.map((n) => n.team);
-
-      if (teamNames) {
-        for (let i=0; i<teams.length; i++) {
-          const index = teamNames.indexOf(teams[i].team);
-          if (index > -1) {
-            if (this._gene?.nominatedtarget) {
-              const nomination = this._gene?.nominatedtarget[index];
-              if (nomination) {
-                nomination.team_data = teams[i];
-                nominations.push(nomination);
-              }
-            }
-          }
-        }
-      }
-      this.nominations = nominations;
+      this.sortNominations(teams);
     });
   }
 
+  sortNominations(teams: Team[]) {
+    if (this._gene && this._gene.nominatedtarget && this._gene.nominatedtarget.length > 0) {
```

Comment:
Could use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) to simplify this, but up to you!

```suggestion
    if (this._gene?.nominatedtarget?.length > 0) {
```